src/sentry/sentry_apps/api/serializers/sentry_app_webhook_request.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from collections.abc import Mapping, MutableMapping, Sequence
+from typing import Any, NotRequired, TypedDict
+
+from sentry.api.serializers import Serializer
+from sentry.hybridcloud.services.organization_mapping import (
+    RpcOrganizationMapping,
+    organization_mapping_service,
+)
+from sentry.sentry_apps.api.utils.webhook_requests import BufferedRequest
+from sentry.sentry_apps.models.sentry_app import SentryApp
+from sentry.users.models.user import User
+from sentry.users.services.user import RpcUser
+
+
+class _BufferedRequestAttrs(TypedDict):
+    organization: RpcOrganizationMapping | None
+
+
+class OrganizationResponse(TypedDict):
+    name: str
+    slug: str
+
+
+class SentryAppWebhookRequestSerializerResponse(TypedDict):
+    webhookUrl: str
+    sentryAppSlug: str
+    eventType: str
+    date: str
+    responseCode: int
+    organization: NotRequired[OrganizationResponse]
+
+
+class SentryAppWebhookRequestSerializer(Serializer):
+    def __init__(self, sentry_app: SentryApp) -> None:
+        self.sentry_app = sentry_app
+
+    def get_attrs(
+        self, item_list: Sequence[BufferedRequest], user: User | RpcUser, **kwargs: Any
+    ) -> MutableMapping[BufferedRequest, _BufferedRequestAttrs]:
+        organization_ids = {item.data.organization_id for item in item_list}
+        organizations = organization_mapping_service.get_many(organization_ids=organization_ids)
+        organizations_by_id = {organization.id: organization for organization in organizations}
+
+        return {
+            item: {
+                "organization": (
+                    organizations_by_id.get(item.data.organization_id)
+                    if item.data.organization_id
+                    else None
+                )
+            }
+            for item in item_list
+        }
+
+    def serialize(
+        self, obj: BufferedRequest, attrs: Mapping[Any, Any], user: Any, **kwargs: Any
+    ) -> SentryAppWebhookRequestSerializerResponse:
+        organization = attrs.get("organization")
+        response_code = obj.data.response_code
+
+        data: SentryAppWebhookRequestSerializerResponse = {
+            "webhookUrl": obj.data.webhook_url,
+            "sentryAppSlug": self.sentry_app.slug,
+            "eventType": obj.data.event_type,
+            "date": obj.data.date,
+            "responseCode": response_code,
+        }
+
+        if organization:
+            data["organization"] = {"name": organization.name, "slug": organization.slug}
```

Comment:
Also we're potentially missing the error [response & request bodies from error webhook requests](https://github.com/getsentry/sentry/blob/master/src/sentry/utils/sentry_apps/request_buffer.py#L160-L177) here

[Same in the RpcModel](https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_apps/services/app_request/serial.py#L7-L12)

Reply:
We decided to remove some properties for now since they're not shown on the dashboard, we can add them back if needed

Reply:
uh can you link me to this discussion ? We shouldn't or we should log them instead since we & customers use that data a lot as a debugging resource

Reply:
hmm, do we have customer stories where they inspect the response we are sending rather than the dashboard?

Reply:
I think the conversation was on Slack but https://github.com/getsentry/sentry/pull/81267#discussion_r1859234013

Reply:
Yes, for this [customer issue](https://github.com/getsentry/sentry/issues/80614) we needed to see our req bodies to validate our webhooks were working. It's also a [customer request in this issue](https://github.com/getsentry/sentry/issues/23879). It's currently not a priority to add the UI to the dashboard but without returning at all to the frontend there is no visibility into the buffer for both customers and us.

Reply:
cc @GabeVillalobos

Reply:
The general guidance I've been giving @ameliahsu is: exclude it until we actually implement this feature. It should be trivial to add to the response body in the future, but I don't want to be exposing this data right now if we aren't using it. It also gives us an opportunity to analyze the overall performance of this endpoint with smaller payloads to start before adding a bunch of additional data.

Reply:
clarification q, is 'this feature' == the endpoint ? or the UI ? Either way though, I still think it's very important to expose that data for debugging. The comment 2 up, sums up the reasons and related issues/asks. Without this data being exposed debugging webhooks & buffer issues becomes incredibly hard without setting up an integration on that sentry app's org.

Reply:
Ah I missed that. Fair point, alright let's go ahead and add it then (sorry @ameliahsu)